### DISCUSSION TITLE
Explorations around exporting to PDF

### DISCUSSION THREAD

**Discussion by Charango-Chile** [quarto-dev/quarto-web#134] [Ideas]
Hi all,

I created a set of revealjs slides using QUARTO with coloured backgrounds, images, some equations and some tweaking of options, some of which led me to ask some questions these previous days which were promptly answered and that I'm grateful for.

There is one thing though that seems to be still very involved and by that I mean exporting to PDF a deck of revealjs slides. Usually, with RMarkdown and Xaringan I used to simply print the presentation in the browser as a PDF and it worked quite well. Well, trying the same with slides created using QUARTO with output to revealjs just creates a royal mess, particularly with the background colours, some image placement and honestly the whole thing looks nowhere near what is being shown on the screen.

In the reveal.js website there is an option mentioned though, which is to install `decktape` and use it to export a deck of slides. It almost work, but then it has two main complications, firstly it gets you out of the usual workspace having to go to a terminal (I can live with it but probably some of the users will not), and secondly the main problem is that depending on the target size you specify, some content will be placed weirdly on the PDF created using this tool... This is particularly true if you use the option `center: true` in your YAML header, but also with some resolutions for the export. I built my deck of slides using 1920 by 1080 as output size but when using the same with `decktape` the results were far form satisfactory, lower resolutions produce better results when it comes to the vertical alignment and placement of some of the material (even if you are using `center; false` in YAML you still get some weirdness).

I know that this is probably a lot to ask, and probably not a highly ranked priority at the moment, but for some of us, creating slides for our units requires to share a PDF version of the same with the students so they follow later and reference to the material contained therein. I know there is a way forward at the moment which requires using `decktape`, which unfortunately is not free of  obstacles, but at least approximates a solution, however, easing this aspect in future QUARTO releases will, in my opinion, increase adoption of QUARTO for some of us that do extensive use of slides decks for lecturing. 

Cheers, Jose

**Comment by jjallaire** [comment]
Printing to PDF w/ Reveal Js should work relatively well (at least at the standard presentation sizes). In case you haven't stumbled across this here are the docs on making it work: https://quarto.org/docs/presentations/revealjs/presenting.html#print-to-pdf

If that doesn't work well for you then this might indeed be too tall an order for the short term. My advice in that case would be to work w/ the revealjs folks to get them to take a PR that improves their print css (which is what we rely on). Here is what they currently use: https://github.com/hakimel/reveal.js/blob/master/css/print/pdf.scss

**Comment by SoaresAlisson** [comment]
After trying with CLI  like `google-chrome-stable --headless --disable-gpu --print-to-pdf=output.pdf --no-pdf-header-footer file:/home/User/complete/path/quarto_report.html`, that worked, but also lacked some basic features, like print in landscape, I arrived in this solution with chromote  https://nanx.me/blog/post/chromote-pdf-automation/ that allows optimization with https://chromedevtools.github.io/devtools-protocol/tot/Page/#method-printToPDF.
